src/Symfony/Bundle/FrameworkBundle/Command/YamlLintCommand.php

```diff
@@ -11,155 +11,36 @@
-use Symfony\Component\Console\Input\InputOption;
-use Symfony\Component\Console\Output\OutputInterface;
-use Symfony\Component\Console\Style\SymfonyStyle;
-use Symfony\Component\Finder\Finder;
-use Symfony\Component\Yaml\Exception\ParseException;
-use Symfony\Component\Yaml\Parser;
+use Symfony\Component\Yaml\Command\LintCommand as BaseLintCommand;
 
 /**
  * Validates YAML files syntax and outputs encountered errors.
  *
  * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Robin Chalas <robin.chalas@gmail.com>
  */
-class YamlLintCommand extends Command
+class YamlLintCommand extends BaseLintCommand
 {
+    /**
+     * {@inheritdoc}
+     */
     protected function configure()
     {
-        $this
-            ->setName('lint:yaml')
-            ->setDescription('Lints a file and outputs encountered errors')
-            ->addArgument('filename', null, 'A file or a directory or STDIN')
-            ->addOption('format', null, InputOption::VALUE_REQUIRED, 'The output format', 'txt')
-            ->setHelp(<<<EOF
-The <info>%command.name%</info> command lints a YAML file and outputs to STDOUT
-the first encountered syntax error.
+        parent::configure();
 
-You can validate the syntax of a file:
+        $this->setHelp(
+                $this->getHelp().<<<EOF
 
-  <info>php %command.full_name% filename</info>
-
-Or of a whole directory:
-
-  <info>php %command.full_name% dirname</info>
-  <info>php %command.full_name% dirname --format=json</info>
-
-Or all YAML files in a bundle:
+Or find all files in a bundle:
 
   <info>php %command.full_name% @AcmeDemoBundle</info>
 
-You can also pass the YAML contents from STDIN:
-
-  <info>cat filename | php %command.full_name%</info>
-
 EOF
-            )
-        ;
-    }
-
-    protected function execute(InputInterface $input, OutputInterface $output)
-    {
-        $io = new SymfonyStyle($input, $output);
-        $filename = $input->getArgument('filename');
-
-        if (!$filename) {
-            if (0 !== ftell(STDIN)) {
-                throw new \RuntimeException('Please provide a filename or pipe file content to STDIN.');
-            }
-
-            $content = '';
-            while (!feof(STDIN)) {
-                $content .= fread(STDIN, 1024);
-            }
-
-            return $this->display($input, $output, $io, array($this->validate($content)));
-        }
-
-        if (0 !== strpos($filename, '@') && !is_readable($filename)) {
-            throw new \RuntimeException(sprintf('File or directory "%s" is not readable', $filename));
-        }
-
-        $files = array();
-        if (is_file($filename)) {
-            $files = array($filename);
-        } elseif (is_dir($filename)) {
-            $files = Finder::create()->files()->in($filename)->name('*.yml');
-        } else {
-            $dir = $this->getApplication()->getKernel()->locateResource($filename);
-            $files = Finder::create()->files()->in($dir)->name('*.yml');
```

Comment:
This would require to define a hard dependency between Yaml and HttpKernel components. Personally I wouldn't mind losing this feature and just be able to check for dirs: `./bin console lint:yaml src/`